modules/sp_dashboard/manifests/init.pp

```diff
@@ -46,47 +33,104 @@
   }
 
-  # Install WSO2 Stream Processor
-  package { $package_name:
-    ensure   => installed,
-    provider => $installer_provider,
-    source   => "/opt/${product}/${product_package}"
+  # Unzip distribution
+  exec { "unpack-jdk":
+    command => "tar -zxvf ${java_home}.tar.gz",
+    path    => "/bin/",
+    cwd     => "${lib_dir}",
+    onlyif  => "/usr/bin/test ! -d ${java_home}",
   }
 
-  # Change the ownership of the installation directory to wso2 user & group
-  file { $install_path:
+  /*
+  * WSO2 Distribution
+  */
+
+  # Create distribution path
+  file { [  "${products_dir}",
+    "${products_dir}/${product}",
+    "${products_dir}/${product}/${product_profile}" ]:
+    ensure => 'directory',
+  }
+
+  # Change the ownership of the installation directory to specified user & group
+  file { $distribution_path:
     ensure  => directory,
     owner   => $user,
     group   => $user_group,
     require => [ User[$user], Group[$user_group]],
     recurse => true
   }
 
-  # Copy deployment.yaml to the installed directory
-  file { "${install_path}/${deployment_yaml_template}":
-    ensure  => file,
-    owner   => $user,
-    group   => $user_group,
-    mode    => '0644',
-    content => template("${module_name}/carbon-home/${deployment_yaml_template}.erb")
+  # Copy binary to distribution path
+  file { "binary":
+    path   => "$distribution_path/${product_binary}",
+    owner  => $user,
+    group  => $user_group,
+    mode   => '0644',
+    source => "puppet:///modules/${module_name}/${product_binary}",
+  }
+
+  # Stop the existing setup
+  exec { "stop-server":
+    command     => "kill -term $(cat ${install_path}/wso2/${product_profile}/runtime.pid)",
+    path        => "/bin/",
+    onlyif      => "/usr/bin/test -f ${install_path}/wso2/${product_profile}/runtime.pid",
+    subscribe   => File["binary"],
+    refreshonly => true,
+  }
+
+  # Wait for the server to stop
+  exec { "wait":
+    command     => "sleep 20",
+    path        => "/bin/",
+    onlyif      => "/usr/bin/test -d ${install_path}",
+    subscribe   => File["binary"],
+    refreshonly => true,
+  }
+
+  # Delete existing setup
+  exec { "detele-pack":
+    command     => "rm -rf ${install_path}",
+    path        => "/bin/",
+    onlyif      => "/usr/bin/test -d ${install_path}",
+    subscribe   => File["binary"],
+    refreshonly => true,
+  }
+
+  # Install the "unzip" package
+  package { 'unzip':
+    ensure => installed,
+  }
+
+  # Unzip the binary and create setup
+  exec { "unzip-update":
+    command     => "unzip -qo ${product_binary}",
+    path        => "/usr/bin/",
+    user        => $user,
+    cwd         => "${distribution_path}",
+    onlyif      => "/usr/bin/test ! -d ${install_path}",
+    subscribe   => File["binary"],
+    refreshonly => true,
+    require     => Package['unzip'],
   }
 
-  # Copy dashboard.sh to installed directory
+  # Copy wso2server.sh to installed directory
```

Comment:
Shouldn't this be dashboard.sh?

Reply:
Fixed https://github.com/wso2/puppet-sp/pull/9/commits/e7029a60e78fa3bc5ece64afaa86ddffd81b11d4